cmdstanpy/utils.py

```diff
@@ -235,15 +239,16 @@ def check_csv(path: str, is_optimizing: bool = False) -> Dict:
     return meta
 
 
-def scan_stan_csv(path: str, is_optimizing: bool = False) -> Dict:
+def scan_stan_csv(path: str, is_sampling: bool = True) -> Dict:
```

Comment:
same comment as above - need to redesign - might need enum on supported methods and scanning routines for each.